checkQC/handlers/qc_handler.py

```diff
@@ -1,81 +1,166 @@
+    has to implement the `collect` method. This method can decide which objects sent to the Subscriber that
+    are of interest to that particular Subscriber, and what should be done with those values.
+    """
+
+    def __init__(self):
+        self.subscriber = self.subscribe()
+        next(self.subscriber)
+
+    def subscribe(self):
+        """
+        This method picks up data from the parser to which the Subscriber is listening.
+        :return:
+        """
+        while True:
+            value = yield
+            self.collect(value)
+
+    def collect(self, signal):
+        """
+        The implementing subclass should provide this method. It is up to instance receiving the data to decide
+        how to handle it. Below is an example of how to handle a tuple with a key-value pair.
+        ```
+        class MySubscriber(Subscriber):
+
+            def __init__(self):
+                self.results = []
+
+            def collect(self, signal):
+                key, value = signal
+                if key == "my_key":
+                    self.results.append(value)
+        ```
+        :return: None
+        """
+        raise NotImplementedError("Implementing class must provide this method.")
+
+    def send(self, value):
+        """
+        Will send the specified value to the subscriber
+        :param value: Value to send to subscriber
+        :return: None
+        """
+        self.subscriber.send(value)
 
 
 class QCHandler(Subscriber):
+    """
+    The QCHandler is one of the fundamental classes of CheckQC. It is the base class for the the concrete
+    implementations of QCHandlers which actually check the the quality criteria of a runfolder.
+    For examples of how to implement a QCHandler it is easiest to look at the implementations available in
+    the `checkQC.handlers` module
+    """
 
     UNKNOWN = 'unknown'
     ERROR = 'error'
     WARNING = 'warning'
 
-    def __init__(self, qc_config, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, qc_config):
+        """
+        Create a QCHandler instance
+        :param qc_config: dict containing the configuration for the QCHandler. Should have keys 'error' and 'warning'
+        """
+        super().__init__()
         self._exit_status = 0
         self.qc_config = qc_config
 
     def custom_configuration_validation(self):
         """
         Override this method in subclass to provide additional configuration behaviour.
         Raise a `ConfigurationError` if there is a problem with the provided config.
+        :raises: ConfigurationError if there is a problem with the configuration
         """
         pass
 
     def validate_configuration(self):
+        """
+        This method will validate the configuration which has been passed to the QCHandler. This should be called
+        by the class making us of this class. It will not be called automatically e.g. at object creation.
```

Comment:
Typo: "us" --> "use"

Reply:
Is it possible to change the first "class" to "sub class" or similar to avoid repeating the same word?